File: 01-Backend/DatajudClient.API/DatajudClient.Domain/DTO/Processos/ReadProcessoDTO.cs
using DatajudClient.Domain.DTO.Enderecos;
using DatajudClient.Domain.DTO.Tribunais;
using System.ComponentModel.DataAnnotations;
using System.Reflection.Metadata.Ecma335;

namespace DatajudClient.Domain.DTO.Processos
{
    public class ReadProcessoDTO
    {
        public int Id { get; set; }
        public string NumeroProcesso { get; set; }
        public string NomeCaso { get; set; }
        public string? Vara { get; set; }
        public string Comarca { get; set; }
        public string? Observacao { get; set; }
        public DateTime UltimoAndamento { get; set; }
        public string UltimoAndamentoDescricao { get; set; }
        public DateTime UltimaAtualizacao { get; set; }
        public ReadEstadoDTO Estado { get; set; }
        public ReadTribunalDTO Tribunal { get; set; }
        public List<ReadAndamentoDTO> Andamentos { get; set; }
    }
}

File: 02-Frontend/DatajudClient.Front/src/app/dtos/processo/readProcessoDto.ts
import { AndamentoProcesso } from "../../interfaces/processo/andamentoProcesso";
import { ReadEstadoDto } from "../endereco/readEstadoDto";
import { ReadCategoriaTribunalDto } from "../tribunal/readCategoriaTribunalDto";
import { ReadTribunalDto } from "../tribunal/readTribunalDto";

export interface ReadProcessoDto {
  Id: number;
  NumeroProcesso: string;
  NomeCaso: string;
  Vara: string | null;
  Comarca: string;
  Observacao: string | null;
  UltimoAndamento: Date;
  UltimaAtualizacao: Date;
  Estado: ReadEstadoDto;
  Tribunal: ReadTribunalDto;
  Andamentos: AndamentoProcesso[];
}
